src/options.cpp

```diff
@@ -8206,8 +8271,10 @@ void options::OnButtonSelectSound(wxCommandEvent& event) {
 
   if (response == wxID_OK) {
     g_sAIS_Alert_Sound_File = g_Platform->NormalizePath(sel_file);
-
-    g_anchorwatch_sound->Stop();
+        g_anchorwatch_sound->Stop();
+#ifdef HAVE_PORTAUDIO
```

Comment:
Why only for portaudio?

Reply:
... because, when you select a new "*.wav" file, you have to reinit portaudio ... an that is the reason .. only for Portaudio. I don't want to change any other sound codes they should work as so far.

Reply:
Note that #ifdef PORTAUDIO doesn't mean that portaudio actually is used -- it's just an indication that it's built. The actual choice of backend is done in the factory, and the only way to know whether we are using portaudio or not is to use a dynamic_cast on the sound pointer.

Reply:
i. e. like this `if (dynamic_cast<SystemCmdSound*>(SoundFactory()))`

Reply:
Yes, I know !
But can you tell me a constalation how it could happen that the the code in Soundfactory.ccp is able to give back another dynamic_cast as one to Portaudio, when HAVE_PORTAUDIO is defined ?
I don't know how this could happen.

... So you could use Portaudio, but you must not, if not defined.

Code :
#include "config.h"
#include "SoundFactory.h"


#ifdef HAVE_PORTAUDIO
#include "PortAudioSound.h"

OcpnSound* SoundFactory(void) { return new PortAudioSound(); }

#elif defined(HAVE_SYSTEM_CMD_SOUND)
#include "SystemCmdSound.h"

OcpnSound* SoundFactory(void) { return new SystemCmdSound(SYSTEM_SOUND_CMD); }

#elif defined(__OCPN__ANDROID__)
#include "AndroidSound.h"

// TBD...

#elif defined(__WXMSW__)
#include "MswSound.h"

OcpnSound* SoundFactory(void) { return new MswSound(); }

#else
#include  "OcpnWxSound.h"

OcpnSound* SoundFactory(void) { return new OcpnWxSound(); }

#endif

Reply:
That should be a simple patch to the factory, a patch which should work. The design is such that it should be possible to change the backend at any point -- and such a change would not necessarily mean that HAVE_PORTAUDIO is undefined.

Besides, #ifdefs are by definition evil should be avoided when possible. They are required to guard code which cannot be compiled on different plaltforms., but there is no reason to use an evil #ifdef when a dynamic_cast makes the job better.